### PR TITLE
hack/jaeger: add elasticsearch backend w kibana

### PR DESCRIPTION
**Pull request by pivotal-jamie-klassen**
# Existing Issue

Related to concourse/concourse#5382.

# Changes proposed in this pull request

* jaeger override now includes elasticsearch & kibana
* as tracing features get expanded, using this override means you can immediately create sophisticated queries/visualizations across the whole span database

# Contributor Checklist
- [x] ~Unit tests~
- [x] ~Integration tests (if applicable)~
- [x] ~Updated documentation (located at https://github.com/concourse/docs)~
- [x] ~Updated release notes (located at https://github.com/concourse/concourse/tree/master/release-notes)~

# Reviewer Checklist
- [x] ~Code reviewed~
- [x] ~Tests reviewed~
- [x] ~Documentation reviewed~
- [x] ~Release notes reviewed~
- [ ] PR acceptance performed
- [x] ~New config flags added? Ensure that they are added to the [BOSH](https://github.com/concourse/concourse-bosh-release) and [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for the [integration tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env) (for example, if they are Garden configs that are not displayed in the `--help` text).~

### CODE REVIEW

File: hack/overrides/jaeger.yml [hunk 1]
```diff
@@ -20,6 +22,9 @@ services:
       - --log-level=debug
     environment:
       COLLECTOR_ZIPKIN_HTTP_PORT: 9411
+      SPAN_STORAGE_TYPE: elasticsearch
+      ES_SERVER_URLS: http://elasticsearch:9200
+      ES_TAGS_AS_FIELDS_ALL: "true"
```
Comment (cirocosta) [inline hack/overrides/jaeger.yml:27]: should we include the change to number of replicas like how we did for hush-house? https://github.com/concourse/hush-house/commit/5b1e70245e540f166222eba12dba2ec732aec590

File: hack/overrides/jaeger.yml [hunk 2]
```diff
@@ -7,6 +7,8 @@
 #
 version: '3'
 
+volumes:
+  elasticsearch-data:
```
Comment (cirocosta) [inline hack/overrides/jaeger.yml:11]: not a big deal, more of a question: is the need of a named volume volume in order to have data being persisted across restarts of elasticsearch?

Comment (pivotal-jamie-klassen) [reply hack/overrides/jaeger.yml:11]: yup, don't want those yellow indices
Screen Shot 2020-04-17 at 9 39 05 AM

File: hack/overrides/jaeger.yml (hunk 1 above)
Comment (pivotal-jamie-klassen) [reply hack/overrides/jaeger.yml:27]: yup, don't want those yellow indices
Screen Shot 2020-04-17 at 9 39 05 AM

File: hack/overrides/jaeger.yml (hunk 2 above)
Comment (pivotal-jamie-klassen) [reply hack/overrides/jaeger.yml:11]: it sure is -- i guess that image doesn't have a volume by default

**Pull request merged**